File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.py
n, k = map(int, input().split())

class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

head = None
tail = None
for i in range(1, n + 1):
    new_node = Node(i)
    if not tail:
        new_node.left = new_node
        new_node.right = new_node
        head = new_node
        tail = new_node
        continue

    new_node.right = head
    new_node.left = tail
    tail.right = new_node
    head.left = new_node
    tail = new_node

temp = head

result = []

while len(result) != n:
    for i in range(k - 1):
        temp = temp.right

    temp.left.right = temp.right
    temp.right.left = temp.left
    result.append(temp.value)
    temp = temp.right

print("<" + ", ".join(map(str, result)) + ">")

File: 백준/Silver/1158. 요세푸스 문제/요세푸스 문제.js
const fs = require('fs');

const [n, k] = fs.readFileSync('/dev/stdin').toString().trim().split(" ").map(Number);

let head
let tail

for(let i = 1; i <= n; i++){
    let newNode = {value: i}
    if (i === 1){
        newNode.left = newNode;
        newNode.right = newNode;
        head = newNode;
        tail = newNode;
        continue;
    }
    
    newNode.left = tail;
    newNode.right = head;
    tail.right = newNode;
    head.left = newNode;
    tail = newNode;
}

const result = []

let temp = head;
while(result.length !== n){
    for(let i = 0; i < k - 1; i++){
        temp = temp.right;
    }
    
    temp.left.right = temp.right;
    temp.right.left = temp.left;
    result.push(temp.value);
    temp = temp.right;
}

console.log("<" + result.map(String).join(", ") + ">")